controllers/cluster_controller.go

```diff
@@ -172,9 +190,34 @@ func (r *ClusterReconciler) reconcile(ctx context.Context, log logr.Logger, clus
 	if err != nil {
 		return ctrl.Result{}, err
 	}
+
+	if reconcileResult.Return() {
+		return reconcileResult.ToCtrlResult(), nil
+	}
+
+	if reconcileResult, err = r.postClusterProviderReconcile(ctx, log, cluster); err != nil {
+		return ctrl.Result{}, err
+	}
+
 	return reconcileResult.ToCtrlResult(), nil
 }
 
+func (r *ClusterReconciler) preClusterProviderReconcile(ctx context.Context, log logr.Logger, cluster *anywherev1.Cluster) (controller.Result, error) {
+	if cluster.HasAWSIamConfig() {
+		return r.awsIamAuth.EnsureCASecret(ctx, log, cluster)
```

Comment:
nit, if you want to have a generic method like this, we shouldn't return directly
```suggestion
		if err := r.awsIamAuth.EnsureCASecret(ctx, log, cluster); err != nil {
		    return err
		}
```